### PR TITLE
LP-1736  Making regflow work with the old update flow

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
### Description

[LP-1736](https://philanthropyu.atlassian.net/browse/LP-1736)

[philu-edx-theme PR](https://github.com/philanthropy-u/philu-edx-theme/pull/1131)

Making regflow work with the old update flow

### Notes

Two extra things other than AC done in this story:

- Criteria for marking user as first learner updated to fix previous bug
		- **This new logic needs to be added in partners reg flow as well, but it will be added when Master is merged into this branch**

- At signup, make user first learner if organisation is orphan

### CODE REVIEW

File: lms/djangoapps/onboarding/models.py [hunk 1]
```diff
@@ -228,6 +228,17 @@ def users_count(self):
...
 
+    def can_join_as_first_learner(self, exclude_user):
+        """
+        Identify, if next user, who will join this organization will become first learner not not.
+
+        :param exclude_user: Currently logged-in user
+        :return: True if this organization is not associated with any
+         user, except currently logged-in user; False otherwise
+        """
+        org_association_count = UserExtendedProfile.objects.filter(organization=self).exclude(user=exclude_user).count()
```
Comment (ahmed-faraz46) [inline lms/djangoapps/onboarding/models.py]: can we use `exists` instead of count?

File: lms/djangoapps/onboarding/views.py [hunk 2]
```diff
@@ -41,6 +42,19 @@
 log = logging.getLogger("edx.onboarding")
 
 
+def check_unattended_surveys(user_extended_profile):
```
Comment (ahmed-faraz46) [inline lms/djangoapps/onboarding/views.py]: There is already a method in UserExtendedProfile `unattended_surveys` which is being used inside this method? can renamed this in a way like it serve the purpose what additional we are doing in it?

Comment (zahid-arbisoft) [reply lms/djangoapps/onboarding/views.py]: Please suggest a name, I have spent sometime and came up with this name

**Comment by saad-waqar** [comment]
LGTM 👍 Great work!

**Pull request merged**